File: raptorsRobotsApp/src/main/java/pl/raptors/raptorsRobotsApp/domain/graphs/Edge.java
package pl.raptors.raptorsRobotsApp.domain.graphs;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@NoArgsConstructor
@Data
@Document(collection = "edges")
public class Edge {

    @Id
    private String id;
    private Vertex vertexA;
    private Vertex vertexB;
    private Boolean biDirected;

    public Edge(Vertex vertexA, Vertex vertexB, Boolean bidirected) {
        this.vertexA = vertexA;
        this.vertexB = vertexB;
        this.biDirected = bidirected;
    }
}

File: angularclient/src/app/model/Graphs/Edge.ts
import { Vertex } from './Vertex';

export class Edge{
  id: string;
  vertexA: Vertex;
  vertexB: Vertex;
  biDirected: boolean;

  constructor(vertexA: Vertex, vertexB: Vertex, biDirected: boolean) {
    this.vertexA = vertexA;
    this.vertexB = vertexB;
    this.biDirected = biDirected;
  }
}
